### PR TITLE
Adds a test for 'checking script type and credentials mode values before connecting to the matched SharedWorkerGlobalScope'

### PR DESCRIPTION
**Pull request by elkurin**
Corresponds to spec PR [#5258](https://github.com/whatwg/html/pull/5258).

This PR adds web-platform-tests to check if a shared worker fires an event named error when it is attempted to connect with mismatch `type` or `credentials`.

### CODE REVIEW

File: workers/shared-worker-options-mismatch.html [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<!DOCTYPE html>
+<title>SharedWorker: type or credentials mismatch failure</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script> setup({allow_uncaught_exception: true}); </script>
```
Comment (annevk) [inline workers/shared-worker-options-mismatch.html]: Why is this needed?

File: workers/shared-worker-options-mismatch.html [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+              { type: 'classic', credentials: 'include' },
+              { type: 'classic' });
+mismatch_test('include to omit in classic',
+              { type: 'classic', credentials: 'include' },
+              { type: 'classic', credentials: 'omit' });
+mismatch_test('include to same-origin in classic',
+              { type: 'classic', credentials: 'include' },
+              { type: 'classic', credentials: 'same-origin' });
+
+// Tests different credentials in module.
```
Comment (annevk) [inline workers/shared-worker-options-mismatch.html]: Are these essentially the same as classic? I'd prefer a `["classic", "module"].forEach` wrapper in that case to make that clear.

File: workers/shared-worker-options-mismatch.html (hunk 1 above)
Comment (elkurin) [reply workers/shared-worker-options-mismatch.html]: Sorry, this is a mistake. I will remove this line.

File: workers/shared-worker-options-mismatch.html (hunk 2 above)
Comment (elkurin) [reply workers/shared-worker-options-mismatch.html]: As you said, the tests for 'module' is same to that of 'classic', so I modified to use forEach wrapper.

**Review by annevk — APPROVED** [review]
Thanks, exhaustive and easy to read!
